src/app/mod.rs

```diff
@@ -257,29 +264,37 @@ where
                 // Otherwise we could reach the prefetch_count and end up blocking
                 // other deliveries if there are a high number of messages with a
                 // future ETA.
-                self.broker.retry(delivery, None).await?;
+                self.broker.ack(&delivery).await?;
```

Comment:
One small change here: I would call `retry` before `ack` in case the `retry` fails for some reason

Reply:
I had chosen this order because if we retry and then ack I think we may have the task executed twice (if `ack` fails). Anyway, in the latest commit I've done it in the order you suggest.